File: Assets/Custom Scripts/csharp/SoundPoolManager.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class SoundPoolManager : MonoBehaviour {
	
	public GameObject audioClipObjectBase;					// AudioClip shell prefab, plz insert
	private int poolSize = 20;								// abitrary poolsize, seemed right
	
	private Stack<GameObject> audioPool;					// our stack of GameObjects to play sound from
	
	// if audio is done, get rid of our Clip
    private	IEnumerator returnClip(AudioClip clip, GameObject obj) 
	{
		yield return new WaitForSeconds(clip.length / obj.audio.pitch);	    
		audioPool.Push(obj);
	}
	
	public void playClipPitchVar(AudioClip clip, Vector3 clipPos, float volume, float pitchVar) 
	{
 		GameObject obj = audioPool.Pop();
															// functions much like PlayClipAtPoint
 		obj.audio.clip = clip;									// except with randomized pitch
  	 	obj.transform.position = clipPos;
   	 	obj.audio.volume = 	volume;
    	obj.audio.pitch = Random.Range(1-pitchVar, 1+pitchVar);	// pass in variance in pitch, either direction														
    	obj.audio.Play();										// play sound
    
		returnClip (clip, obj);
	}
	
	void Start () 
	{		
		for (int counter = 0; counter == poolSize ; counter++) {			// makin a bunch of empty GameObjects to use 
			GameObject newShell = GameObject.Instantiate(audioClipObjectBase) as GameObject;		// as a pool for playing sounds with pitch
			audioPool.Push(newShell);
		}
	}

}

File: Assets/soundPoolManager.js
import System.Collections.Generic;

var audioPool : List.<GameObject> = new List.<GameObject>();
var audioBase : GameObject;
private var poolSize : int = 20;									// 20 sounds at once

var poolObj : GameObject;
var picoPops : AudioClip[];

function getChildObjects(rootObj : String, childString: String) {

}


function Start () {
	
	var aTransform : Transform;
	
	var poolContainer = GameObject.Find("Sound Pool");
	
	var audioCCs = GetComponentsInChildren(AudioSource);
	var counter : int;
	for (counter = 0; counter <= poolSize ; counter++) {			// makin a bunch of empty GameObjects to use 
		var newShell = GameObject.Instantiate(audioBase);			// as a pool for playing sounds with pitch
		audioPool.Add(newShell);
	}
}

function notPlaying(GO : GameObject) {
	return (GO.audio.isPlaying == false);
}

function getClipObj() { 
	var index = Random.Range(0, audioPool.Count);
	
	var obj =  audioPool[index];
	return obj;	
}

function randPitch(a : AudioClip){

}

function playPitchedClip(clip : AudioClip, clipPos: Vector3, volume : float, pitchVar: float) {
 	var obj = getClipObj();
															// functions much like PlayClipAtPoint
 	obj.audio.clip = clip;									// except with randomized pitch
    //obj.transform.position = clipPos;	
    obj.transform.position = gameObject.transform.position;
    obj.audio.volume = 	volume;
    obj.audio.pitch = Random.Range(1-pitchVar, 1+pitchVar);	// pass in variance in pitch, either direction														
    obj.audio.Play();										// play sound
    
    yield WaitForSeconds((clip.length / obj.audio.pitch));	    
	// if audio is done, get rid of our Clip
    audioPool.Add(obj);
	
}